packages/mui-base/src/useSwitch/useSwitch.ts

```diff
@@ -101,6 +101,46 @@ export default function useSwitch(props: UseSwitchParameters): UseSwitchReturnVa
 
   const handleInputRef = useForkRef(focusVisibleRef, inputRef);
 
+  const createHandleClick =
+    (otherProps: React.InputHTMLAttributes<HTMLInputElement>) =>
+    (event: React.MouseEvent<HTMLInputElement>) => {
+      setCheckedState((checkedState) => !checkedState);
+      setFocusVisible(false);
+      otherProps.onClick?.(event);
+
+      const nativeEvent = event.nativeEvent || event;
+      // @ts-ignore
+      const clonedEvent = new nativeEvent.constructor(nativeEvent.type, nativeEvent);
+
+      Object.defineProperty(clonedEvent, 'target', {
+        writable: true,
+        value: { type: 'checkbox', checked: !checked },
+      });
+      onChange?.(clonedEvent);
+      otherProps.onChange?.(clonedEvent);
+    };
+
+  const createHandleKeyDown =
+    (otherProps: React.InputHTMLAttributes<HTMLInputElement>) =>
+    (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.code === 'Space' && inputRef.current) {
+        event.preventDefault();
+        setCheckedState(!checked);
+        otherProps.onKeyDown?.(event);
+
+        const nativeEvent = event.nativeEvent || event;
+        // @ts-ignore
+        const clonedEvent = new nativeEvent.constructor(nativeEvent.type, nativeEvent);
+
+        Object.defineProperty(clonedEvent, 'target', {
+          writable: true,
+          value: { type: 'checkbox', checked: !checked },
+        });
+        onChange?.(clonedEvent);
+        otherProps.onChange?.(clonedEvent);
+      }
+    };
+
   const getInputProps: UseSwitchReturnValue['getInputProps'] = (otherProps = {}) => ({
```

Comment:
It seems that this function is not used anymore. I'd still use it to set the `type` and the `name` attribute. It could also be useful to add a test verifying if the `name` is set correctly and the switch's value is considered when a parent form is submitted.

Reply:
![image](https://github.com/mui/material-ui/assets/56389521/f8bd5dc0-9aa3-4045-bc3f-b8166e48afb4)
isn't this an example of `getInputProps` being used? It's from `packages/mui-joy/build/modern/Switch/Switch.js`

Reply:
@michaldudak Sorry I don't really understand. What do you mean by setting `type` and `name`? Should the switch component be able to receive `type` and `name` props?

Reply:
In the end I changed nothing from `getInputProps` but I've added a test to test wether props given to input Slot is forwarded to the input

Reply:
> Should the switch component be able to receive type and name props?

`name` - yes, `type` - no

`type` should be set to `hidden`. If it's not set, it defaults to `text`, which is not what we want.

As for `name`, we can forward the `name` prop from the `Switch` to its `input` slot, so it's easier to use: `<Switch name="my-switch" />` will render `<span ...><!-- ... --><input type="hidden" name="my-switch" /></span>`

The test you added is not necessary. This logic is already being tested in `describeConformanceUnstyled` test suite.

Reply:
Thank you for the explanation. I've added `type` directly to the component and `name` through `additionalProps`. Added also the corresponding test. I don't add it through `getInputProps` since I think this props is special for `Switch`, so that other use case of `getInputProps` don't get unnecesary data. Please let me know if you want it through `getInputProps` instead.

Reply:
@michaldudak ready for a re-review

Reply:
> I don't add it through getInputProps since I think this props is special for Switch, so that other use case of getInputProps don't get unnecesary data.

What do you mean by that? The `getInputProps` result is meant to be applied on an input that belongs to a Switch. The `type` does belong there.